plugins/modules/cloudwatch_metric_alarm.py

```diff
@@ -340,6 +471,14 @@ def main():
     params['InsufficientDataActions'] = module.params.get('insufficient_data_actions', [])
     params['OKActions'] = module.params.get('ok_actions', [])
     params['TreatMissingData'] = module.params.get('treat_missing_data')
+    if module.params.get('metrics'):
+        params['Metrics'] = snake_dict_to_camel_dict(module.params['metrics'], capitalize_first=True)
+    if module.params.get('extended_statistic'):
+        params['ExtendedStatistic'] = module.params.get('extended_statistic')
+
+    for key, value in list(params.items()):
```

Comment:
Why use `list()` here?

Reply:
list() is used here because it preserves the keys & values as they appeared when the list was created. If not, the usage of `del` inside the loop will result in `RuntimeError: dictionary changed size during iteration`

Reply:
Ah, right. You could use a dictionary comprehension here which would be a little more pythonic, but either way is fine.

Reply:
Why are you deleting None values?

Reply:
Why are you deleting None values? I guess it would be easier to set a params only if defined instead. Otherwise, you have to cover that functionality with a unit test.

Reply:
@alinabuzachis The keys that are not defined in the task will be set to 'None' value. I am deleting the None values in order to satisfy the mutually exclusive condition. I will add a test case to test the mutually exclusive condition to the integration test suite, for now.